tests/components/figure.test.ts

```diff
@@ -238,6 +239,29 @@ describe("figures", () => {
     }
   );
 
+  test.each([
+    ["right", { mouseOffsetX: 300, mouseOffsetY: 0 }],
+    ["bottom", { mouseOffsetX: 0, mouseOffsetY: 300 }],
+    ["bottomRight", { mouseOffsetX: 300, mouseOffsetY: 300 }],
+  ])(
+    "Resizing a figure through its right and bottom anchor works as expected when viewport is scrolled and figure is partially visible",
```

Comment:
```suggestion
    "Resizing a figure does not crop it to its visible part in the viewport",
```

Reply:
@rrahir, I've made the changes, Thanks :)